### ISSUE TITLE
FAKE Build failing on Mac

### ISSUE THREAD

**Issue by patridge** [fsharp/FAKE#696]
## Issue

When I tried to build FAKE on my Mac after making some changes to a helper, it coughed up some warning messages and then fail to build. I don't know if I'm missing some pre-requisite installed item(s), but I do have Xamarin Studio installed with its F# support. I also tried specifically installing Mono for Mac outside XS.

Here are the types of warnings I see for most of the projects in FAKE.sln and FAKE.Deploy.Web.sln:

> /Users/me/Projects/FAKE/src/app/FakeLib/FakeLib.fsproj:  warning : Could not find project file /usr/local/Cellar/mono/3.10.0/lib/mono/Microsoft SDKs/F#//Framework/v4.0/Microsoft.FSharp.Targets, to import. Ignoring.

...which lead to errors like these:

> /Users/me/Projects/FAKE/src/app/FakeLib/FakeLib.fsproj: error : /Users/me/Projects/FAKE/src/app/FakeLib/FakeLib.fsproj: /Users/me/Projects/FAKE/src/app/FakeLib/FakeLib.fsproj could not import "$(FSharpTargetsPath)"
/Users/me/Projects/FAKE/src/app/FAKE/FAKE.fsproj: error : /Users/me/Projects/FAKE/src/app/FAKE/FAKE.fsproj: The required attribute "Project" in Import is empty

## Potential Solution

On my particular OS X system (10.10.2), the F# SDKs are not in `/usr/local/...`, but are in `/Library/Frameworks/Mono.framework/Versions/Current/lib/mono/Microsoft SDKs/F#` instead. Once I found the project XML in question, I tried to adapt it to handle both the original and my Library locations using logic similar to what is there to detect the 3.0/3.1 version.

    <PropertyGroup>
      <FSharpSdksLocation Condition="Exists('\Library\Frameworks\Mono.framework\Versions\Current\lib\mono\Microsoft SDKs\F#')">/Library/Frameworks/Mono.framework/Versions/Current/lib/mono/Microsoft SDKs/F#</FSharpSdksLocation>
      <FSharpSdksLocation Condition="Exists('$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#')">$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#</FSharpSdksLocation>
      <FSharpSdkVer Condition="Exists('$(FSharpSdksLocation)\3.0\Framework\v4.0\Microsoft.FSharp.Targets')">3.0</FSharpSdkVer>
      <FSharpSdkVer Condition="Exists('$(FSharpSdksLocation)\3.1\Framework\v4.0\Microsoft.FSharp.Targets')">3.1</FSharpSdkVer>
      <FSharpTargetsPath>$(FSharpSdksLocation)\$(FSharpSdkVer)\Framework\v4.0\Microsoft.FSharp.Targets</FSharpTargetsPath>
    </PropertyGroup>
    <Import Project="$(FSharpTargetsPath)" />

One thing this does is normalize these PropertyGroup/Import calls in all those projects. Some were only [conditionally importing if it exists](https://github.com/fsharp/FAKE/blob/master/src/app/FAKE/FAKE.fsproj#L55) (possibly hiding an import error with build errors). Some were doing the conditional version logic with `FSharpSdkVer` within a single `PropertyGroup` like above and some were [doing separate PropertyGroups](https://github.com/fsharp/FAKE/blob/master/src/app/FAKE/FAKE.fsproj#L49).

To see the changes so far, [check out this commit in my branch](https://github.com/patridge/FAKE/commit/07a19c72a1aed3ee7666c5adb3baa93e4a415cfd).

Since I don't trust my project XML skills, I'm just going to submit this as an issue rather than a pull request, especially since it isn't fully building yet.

## Remaining Warnings/Errors

After [these changes](https://github.com/patridge/FAKE/commit/07a19c72a1aed3ee7666c5adb3baa93e4a415cfd), I can build FAKE.sln successfully (with a few non-critical warnings), but FAKE.Deploy.Web.sln dies with similar import warnings/errors as above.

Warnings:

> /Users/me/Projects/FAKE/FAKE.Deploy.Web.sln:  warning : /Users/me/Projects/FAKE/src/test/Test.FAKECore/Test.FAKECore.csproj: ProjectReference '..\..\app\FAKE\FAKE.fsproj' not found, neither by guid 'ca923bca-c273-4744-9658-7df7001ff266' nor by project file name '/Users/me/Projects/FAKE/src/app/FAKE/FAKE.fsproj'.
/Users/me/Projects/FAKE/FAKE.Deploy.Web.sln:  warning : /Users/me/Projects/FAKE/src/deploy.web/Fake.Deploy.Web.DataProviders/Fake.Deploy.Web.RavenDb/Fake.Deploy.Web.RavenDb.fsproj: The required attribute "Project" in Import is empty
/Users/me/Projects/FAKE/src/test/Test.Fake.Deploy.Web.File/Test.Fake.Deploy.Web.File.fsproj:  warning : Cannot import project '/Library/Frameworks/Mono.framework/Versions/Current/lib/mono/Microsoft SDKs/F#/3.1/Framework/v4.0/Microsoft.FSharp.Targets' again. It was already imported by '/Users/me/Projects/FAKE/src/test/Test.Fake.Deploy.Web.File/Test.Fake.Deploy.Web.File.fsproj'. Ignoring.
/Users/me/Projects/FAKE/src/test/Test.Fake.Deploy.Web/Test.Fake.Deploy.Web.fsproj:  warning : Cannot import project '/Library/Frameworks/Mono.framework/Versions/Current/lib/mono/Microsoft SDKs/F#/3.1/Framework/v4.0/Microsoft.FSharp.Targets' again. It was already imported by '/Users/me/Projects/FAKE/src/test/Test.Fake.Deploy.Web/Test.Fake.Deploy.Web.fsproj'. Ignoring.

Errors:

> /Users/me/Projects/FAKE/src/deploy.web/Fake.Deploy.Web.DataProviders/Fake.Deploy.Web.RavenDb/Fake.Deploy.Web.RavenDb.fsproj: error : /Users/me/Projects/FAKE/src/deploy.web/Fake.Deploy.Web.DataProviders/Fake.Deploy.Web.RavenDb/Fake.Deploy.Web.RavenDb.fsproj: The required attribute "Project" in Import is empty

**Comment by patridge** [comment]
On a happy note, `build.cmd` from my branch on Windows still built just fine with these changes in place.

**Comment by haf** [comment]
Related https://github.com/fsharp/fsharp/issues/378

**Comment by dsyme** [comment]
@patridge Is this fixed for you?

**Comment by patridge** [comment]
@dsyme It appears that all is well these days. Thank you for checking.

I stopped using FAKE shortly after filing this issue (and apparently deleted the referenced fork/branch at some point since). I'm no longer on the same machine, either. That said, I just downloaded the latest source after receiving your message and did a full `./build.sh`. Everything built fine on this newer Mac and Mono version (10.11.4 and 4.2.3, respectively).
